### PR TITLE
fix: :bug: 1104 health factor

### LINKED ISSUE #1104: Health Factor is showing as a negative number in some cases in repay modal
**Describe the bug**
In the repay modal, the future health factor number is calculated as a negative number in some cases, which is confusing in the UI.

**Screenshots**
image

**Additional context**
There was a similar issue with the remaining debt value showing as negative. That was fixed by setting it to 0 if it was negative, so a similar fix might be needed here.

This issue probably exists in both the repay with wallet balance and the repay with collateral modals, so the fix might have to be applied in both spots (if that logic isn't shared already)

### PR DESCRIPTION
**Pull request by lyzergic**
Fixed health factor bug where it showed bellow zero values

## General Changes

- closes #1104 bug

## Developer Notes

None

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [x]  The base branch is set to `main`
- [x]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [x]  The Github issue has been linked to the PR in the Development section
- [x]  The General Changes section has been filled out
- [x]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [x]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Review by defispartan — APPROVED** [review]

File: src/components/transactions/Repay/RepayModalContent.tsx [hunk 1]
```diff
@@ -203,11 +203,13 @@ export const RepayModalContent = ({
...
 
-    newHF = calculateHealthFactorFromBalancesBigUnits({
+    const calculatedHealthFactor = calculateHealthFactorFromBalancesBigUnits({
       collateralBalanceMarketReferenceCurrency,
       borrowBalanceMarketReferenceCurrency,
       currentLiquidationThreshold: user?.currentLiquidationThreshold || '0',
-    }).toString(10);
+    });
+
+    newHF = calculatedHealthFactor.isLessThan(0) ? '0' : calculatedHealthFactor.toString(10);
```
Comment (defispartan) [inline src/components/transactions/Repay/RepayModalContent.tsx:212]: `calculatedHealthFactor` can equal -1 if the full balance is repaid, which shouldn't be reset, so could you change the condition:

`calculatedHealthFactor.isLessThan(0) && !calculatedHealthFactor.eq(-1)`

File: src/utils/hfUtils.ts [hunk 2]
```diff
@@ -158,6 +158,6 @@ export const calculateHFAfterRepay = ({
 
   return {
     hfEffectOfFromAmount: valueToBigNumber(hfInitialEffectOfFromAmount),
-    hfAfterSwap,
+    hfAfterSwap: hfAfterSwap.isLessThan(0) ? 0 : hfAfterSwap,
```
Comment (defispartan) [inline src/utils/hfUtils.ts:161]: Same here, 
`hfAfterSwap.isLessThan(0) && !hfAfterSwap.eq(-1)`

Comment (lyzergic) [reply src/utils/hfUtils.ts]: done!

File: src/components/transactions/Repay/RepayModalContent.tsx (hunk 1 above)
Comment (lyzergic) [reply src/components/transactions/Repay/RepayModalContent.tsx]: done!

**Review by defispartan — APPROVED** [review]

**Review by JoaquinBattilana — APPROVED** [review]

**Pull request merged**